File: 2029-stone-game-ix/2029-stone-game-ix.cpp
class Solution {
public:
    bool stoneGameIX(vector<int>& stones) {
        int f[3] = {0, 0, 0};
        for (auto& s : stones)
            f[s % 3]++;
        if (~f[0] & 1)
            return min(f[1], f[2]) >= 1;
        return abs(f[1] - f[2]) >= 3;
    }
};

File: 2029-stone-game-ix/2029-stone-game-ix.java
class Solution {
    public boolean stoneGameIX(int[] stones) {
        int[] f = {0, 0, 0};
        for (int s : stones)
            f[s % 3]++;
        if ((f[0] & 1) == 0)
            return Math.min(f[1], f[2]) > 0;
        return Math.abs(f[1] - f[2]) > 2;
    }
}